### ISSUE TITLE
Captive Portal attack hangs on AP start.

### ISSUE THREAD

**Issue by SwiftPredator** [FluxionNetwork/fluxion#92]
### Do you want to request a *feature* or report a *bug*?
-> *Bug*
As mentioned before to @MPX4132  the Captive Portal freezes after starting. 

![virtualbox_kali-linux-2017 1-vbox-amd64_19_08_2017_00_27_41](https://user-images.githubusercontent.com/20613170/29480550-5a3f8946-8479-11e7-8519-a1259b842339.png)

**Comment by dreni038** [comment]
Same is happening to me
attacks/Captive Portal/attack.sh: line 175: /root/fluxion/attacks/Captive Portal/sites/generic/*: No such file or directory
[*] Starting Captive Portal access point service...
It says this and doesnt do anything

**Comment by MPX4132** [comment]
@dreni038 Yours is different, it just means you need to download the sites. They were recently moved to a submodule. Get them from [here](https://github.com/FluxionNetwork/sites/tree/22a2cafc3d8272a39bc4efcec0b14ba5b75cd55b) and save them to "fluxion/attacks/Captive Portal/sites".

**Comment by SwiftPredator** [comment]
@MPX4132  You need more informations or do you know what the problem could be?

**Comment by dreni038** [comment]
@MPX4132 Thank you.. i just got them and put them in the library.. i hope that it will work now..

**Comment by MPX4132** [comment]
Make sure there's a "generic" directory in sites, so it's it must be "fluxion/attacks/Captive Portal/sites/generic"

**Comment by dreni038** [comment]
@MPX4132 Yes there is a generic folder in there.. im just waiting for a handshake so i can see if it works now or not.. I apologize since i am a beginner..

**Comment by MPX4132** [comment]
It's all good.

**Comment by dreni038** [comment]
Now it ends at "[*] Starting Captive Portal access point service..." and opens a xterm terminal saying "Can't execvp hostpad: No such file or directory" any ideas?

**Comment by MPX4132** [comment]
Can you provide a screen shot or copy and paste the output of all windows? I haven't seen that bug before.

**Comment by MPX4132** [comment]
You may not have xterm installed, x-terminal-emulator is not xterm.

**Comment by MPX4132** [comment]
@SwiftPredator Sorry, I hadn't seen your reply. I meant to link (this other issue)[https://github.com/FluxionNetwork/fluxion/issues/73] earlier. Follow the steps I put there to see what happens.

**Comment by SwiftPredator** [comment]
So after following the first steps(Trying airbase-ng) i can say that airbase gave me the same problem. Maybe a RTL8812AU driver issue? I think i will never bring fluxion to work on this chipset :/

**Comment by SwiftPredator** [comment]
@MPX4132  here is the output when i start it not in Fluxion.

Configuration file: /etc/mana-toolkit/hostapd-mana.conf
nl80211 driver initialization failed.
hostapd_free_hapd_data: Interface wlan0 wasn't started
Hit enter to kill me

**Comment by SwiftPredator** [comment]
Just for the info. I started hostapd manuel multiple times and one time it worked and hosted a network, but the other times i had the above erroer

**Comment by SwiftPredator** [comment]
So, here is a screenshot showing the bug really nice. On the left terminal you can see that mana-hostapd worked before starting fluxion. I started fluxion and it froze on the hosted part(same with airbase). After closing fluxion, also the mana-hostapd didn't worked anymore.
![virtualbox_kali-linux-2017 1-vbox-amd64_19_08_2017_04_14_36](https://user-images.githubusercontent.com/20613170/29482628-4856c19c-8495-11e7-94a7-43c0280eda37.png)

**Comment by MPX4132** [comment]
Just to verify it in fact isn't a configuration problem, try this:
Pick an interface you want to use and set it manually in monitor mode:
```
ip link set wlanX down
iwconfig wlanX mode monitor
ip link set wlanX up
```
Alternatively, try:
```
ifconfig wlanX down
iwconfig wlanX mode monitor
ifconfig wlanX up
```
Check the interface is in monitor mode with:
```
iwconfig wlanX
```
It should say "Mode:Monitor" and not "Mode:Managed."

Now, create a test config file:
```
interface=wlanX
driver=nl80211
ssid=TestAP
channel=1
```
Save it as hostapd.conf
Finally, try starting hostapd with it:
```
hostapd hostapd.conf
```
You should get a screen saying:
```
Configuration file: hostapd.conf
Using interface wlan1mon with hwaddr XX:XX:XX:XX:XX:XX and ssid "TestAP"
wlan1mon: interface state UNINITIALIZED->ENABLED
wlan1mon: AP-ENABLED
```
If this looks fine, check with another device (like a phone/tablet) if a new AP is available.
The AP will not have any routes, but it should spawn and you should see it available.

**Comment by dreni038** [comment]
Sorry,i was sleeping.. Yeah it seems i havent installed xterm and im using xterminal emulator.. i tried running th apt-get command  but it doesnt work..any ideas?

**Comment by SwiftPredator** [comment]
Hey @MPX4132 ,

here is the output: 

 hostapd hostapd.conf
Configuration file: hostapd.conf
Using interface wlan0 with hwaddr 40:a5:ef:0d:19:70 and ssid "TestAP"
wlan0: interface state UNINITIALIZED->ENABLED
wlan0: AP-ENABLED 

No Ap is shown with this Method. I saved the hostapd.conf file in the root folder. With Mana-toolkit a AP was shown on my other devices.

**Comment by SwiftPredator** [comment]
with this command it works fine. A new AP called "Internet" is shown on all my devices.

bash /usr/share/mana-toolkit/run-mana/start-nat-simple.sh 
Configuration file: /etc/mana-toolkit/hostapd-mana.conf
Using interface wlan0 with hwaddr 00:11:22:33:44:00 and ssid "Internet"
wlan0: interface state UNINITIALIZED->ENABLED
wlan0: AP-ENABLED 
Hit enter to kill me

**Comment by MPX4132** [comment]
@SwiftPredator Can you post what start-nat-simple.sh does (the script's code), along with the contents of the hostapd-mana.conf file? @dreni038 I'm not really sure about that, you'll have to figure that one out yourself. You could also open another issue here, maybe someone else that's solve that problem can help you out.

**Comment by SiNaPsEr0x** [comment]
some problem on run hostapd with ar9271... this log
p: impossibile eseguire stat di '/root/hack/fluxion/attacks/Captive Portal/sites/*_*.portal': File o directory non esistente
find: ‘/tmp/fluxspace/captive_portal/’: File o directory non esistente
Captive Portal authentication mode is already set, skipping!
Captive Portal certificate is already set, skipping!
rm: impossibile rimuovere '/tmp/fluxspace/captive_portal': File o directory non esistente

**Comment by SiNaPsEr0x** [comment]
please add sites to default git or download automatic

**Comment by MPX4132** [comment]
You should open another issue with that, @deltaxflux changed it recently.

**Comment by SwiftPredator** [comment]
Here the content of start-nat-simple.sh:

#!/bin/bash

upstream=eth0
phy=wlan0
conf=/etc/mana-toolkit/hostapd-mana.conf
hostapd=/usr/lib/mana-toolkit/hostapd

service network-manager stop
rfkill unblock wlan

ifconfig $phy up

sed -i "s/^interface=.*$/interface=$phy/" $conf
$hostapd $conf&
sleep 5
ifconfig $phy 10.0.0.1 netmask 255.255.255.0
route add -net 10.0.0.0 netmask 255.255.255.0 gw 10.0.0.1

dnsmasq -z -C /etc/mana-toolkit/dnsmasq-dhcpd.conf -i $phy -I lo

echo '1' > /proc/sys/net/ipv4/ip_forward
iptables --policy INPUT ACCEPT
iptables --policy FORWARD ACCEPT
iptables --policy OUTPUT ACCEPT
iptables -F
iptables -t nat -F
iptables -t nat -A POSTROUTING -o $upstream -j MASQUERADE
iptables -A FORWARD -i $phy -o $upstream -j ACCEPT

echo "Hit enter to kill me"
read
pkill dnsmasq
pkill sslstrip
pkill sslsplit
pkill hostapd
pkill python
iptables -t nat -F

Here the content of hosted-mana.conf:

#A full description of options is available in https://github.com/sensepost/hostapd-mana/blob/master/hostapd/hostapd.conf 

interface=wlan0
bssid=00:11:22:33:44:00
driver=nl80211
ssid=Internet
channel=6

# Prevent dissasociations
disassoc_low_ack=0
ap_max_inactivity=3000

# Both open and shared auth
auth_algs=3

# no SSID cloaking
#ignore_broadcast_ssid=0

# -1 = log all messages
logger_syslog=-1
logger_stdout=-1

# 2 = informational messages
logger_syslog_level=2
logger_stdout_level=2

ctrl_interface=/var/run/hostapd
ctrl_interface_group=0

# Finally, enable mana
enable_mana=1
# Limit mana to responding only to the device probing (0), or not (1)
mana_loud=0
# Extend MAC ACLs to probe frames
mana_macacl=0
# Put hostapd in white/black list mode
#macaddr_acl=0
# only used if you want to do filter by MAC address
#accept_mac_file=/etc/mana-toolkit/hostapd.accept
#deny_mac_file=/etc/mana-toolkit/hostapd.deny

**Comment by MPX4132** [comment]
@SwiftPredator please run the following command and post the output here. 
```
which hostapd
```

**Comment by SwiftPredator** [comment]
/usr/sbin/hostapd

**Comment by MPX4132** [comment]
There's the problem... You're using two different hostapd to test... One's the real hostapd (/usr/sbin/hostapd), the other is a custom hostapd (/usr/lib/mana-toolkit/hostapd)

**Comment by SwiftPredator** [comment]
What should i do o fix it  ?

**Comment by MPX4132** [comment]
Well, it looks like the real hostapd doesn't work for you. The custom one appears to work sometimes. The problem stems from what seems to be an issue with the driver; it's unstable. This is going to be the last I can help you with. You can modify fluxion/lib/hostapd.sh and change the line that starts hostapd to use your custom one. Change this line at the very bottom of the file: 
```
xterm $HOLD $BOTTOMRIGHT -bg "#000000" -fg "#FFFFFF" -title "FLUXION AP Service [hostapd]" -e hostapd "$APServiceConfigDirectory/$APRogueMAC-hostapd.conf" &
```
to your custom version of hostapd
```
xterm $HOLD $BOTTOMRIGHT -bg "#000000" -fg "#FFFFFF" -title "FLUXION AP Service [hostapd]" -e /usr/lib/mana-toolkit/hostapd "$APServiceConfigDirectory/$APRogueMAC-hostapd.conf" &
```

**Comment by SwiftPredator** [comment]
Didnt worked out :( The problem here is, if it don't work with my RTL8812AU Chipset it won't work with all of them. So the airmon-ng independence is useless for those typ of Chipset. If you maybe get a solution in the near future, i would be really happy.

Thanks for your help !

**Comment by SwiftPredator** [comment]
Ah, and starting airbase-ng manual works also:

airbase-ng wlan0
19:05:48  Created tap interface at0
19:05:48  Trying to set MTU on at0 to 1500
19:05:48  Trying to set MTU on wlan0 to 1800
19:05:48  Access Point with BSSID F2:AB:E3:0E:00:5F started.

**Comment by SwiftPredator** [comment]
After some time this happens. Maybe just driver problems. I don't know. Still Thanks for your help.

read failed: Network is down
wi_read(): Network is down
read failed: Network is down
wi_read(): Network is down

**Comment by SiNaPsEr0x** [comment]
its all ok now with ar9271 chip with hostapd and kali 4.11
